src/main/java/com/cognifide/aemrules/checks/SessionShouldBeLoggedOut.java

```diff
@@ -58,16 +68,51 @@ public void scanFile(JavaFileScannerContext javaFileScannerContext) {
 
 	@Override
 	public void visitMethod(MethodTree method) {
-		Collection<VariableTree> sessions = findSessionsInMethod(method);
-		for (VariableTree session : sessions) {
-			boolean closed = checkIfLoggedOut(method, session);
-			if (!closed) {
-				context.reportIssue(this, session, RULE_MESSAGE);
+		if (!checkIfLongSession(method)) {
+			Collection<VariableTree> sessions = findSessionsInMethod(method);
+			for (VariableTree session : sessions) {
+				boolean closed = checkIfLoggedOut(method, session);
+				if (!closed) {
+					context.reportIssue(this, session, RULE_MESSAGE);
+				}
 			}
 		}
 		super.visitMethod(method);
 	}
 
+	protected boolean checkIfLongSession(MethodTree method) {
+		List<AnnotationTree> annotations = method.modifiers().annotations();
+		for (AnnotationTree annotationTree : annotations) {
+			if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER)) {
+				IdentifierTree idf = (IdentifierTree) annotationTree.annotationType();
+				if (idf.name().equals(ACTIVATE)) {
+					checkIfLongSessionOpened(method);
+					return true;
+				}
+				else if (idf.name().equals(DEACTIVATE)) {
```

Comment:
Will it work, if deactivate method will go before activate method in class? just asking :-)